File: python/001_get_connection_create_document.py
from mapr.ojai.storage.ConnectionFactory import ConnectionFactory

# Create a connection to data access server
connection_str = "localhost:5678?auth=basic;user=mapr;password=mapr;" \
          "ssl=true;" \
          "sslCA=/opt/mapr/conf/ssl_truststore.pem;" \
          "sslTargetNameOverride=node1.mapr.com"
connection = ConnectionFactory.get_connection(connection_str=connection_str)

# Json string or json dictionary
json_dict = {"_id": "id001",
             "name": "Joe",
             "age": 50,
             "address": {
                 "street": "555 Moon Way",
                 "city": "Gotham"}
             }

# Create new document from json_document
new_document = connection.new_document(dictionary=json_dict)

# Print the OJAI Document
print(new_document.as_json_str())

# close the OJAI connection
connection.close()

File: csharp/001_GetConnectionCreateDocument.cs
using System;
using MapRDB.Driver;

public class GetConnectionCreateDocument
{
    public void GetConnectionCreateDocument()
    {
        // Create a connection to data access server
        var connectionStr = $"localhost:5678?auth=basic;" +
            $"user=mapr;" +
            $"password=mapr;" +
            $"ssl=true;" +
            $"sslCA=/opt/mapr/conf/ssl_truststore.pem;" +
            $"sslTargetNameOverride=node1.mapr.com";
        var connection = ConnectionFactory.CreateConnection(connectionStr);

        // Json string
        var jsonStr =
            @"{" +
                @"""_id"":""id001""," +
                @"""name"":""Joe""," +
                @"""age"":{""$numberInt"":""50""}," +
                @"""address"":" +
                    @"{" +
                        @"""street"":""555 Moon Way""," +
                        @"""city"":""Gotham""" +
                    @"}" +
            @"}";

        // Create a document from jsonStr
        var documentJson = connection.NewDocument(jsonStr);

        // Print the OJAI Document
        Console.WriteLine(documentJson.ToJsonString());

        // Create new document with the same fields using constructor
        var documentConstructed = connection.NewDocument()
            .SetID("id001")
            .Set("name", "Joe")
            .Set("age", 50)
            .Set("address.street", "555 Moon Way")
            .Set("address.city", "Gotham");

        // Print the OJAI Document
        Console.WriteLine(documentConstructed.ToJsonString());

        // Close the OJAI connection
        connection.Close();
    }
}
